test_common/harness/kernelHelpers.cpp

```diff
@@ -614,6 +614,26 @@ static int get_offline_compiler_output(
             }
         }
     }
+
+    if (compilationMode == kSpir_v && !gDisableSPIRVValidation)
+    {
+        std::string runString = gSPIRVValidator + " " + outputFilename;
+
+        int returnCode = system(runString.c_str());
+        if (returnCode == -1)
+        {
+            log_error("Error: failed to invoke SPIR-V validator\n");
+            return CL_COMPILE_PROGRAM_FAILURE;
```

Comment:
This will fail the CTS test if we can't find the validator or if it returns an error code.  Is this the intended behavior?  Debatably this should more of a "warning" than an "error", since it's more of a test of the SPIR-V compiler than the OpenCL implementation under test.

Reply:
The behavior is as intended. @kpet, would you care to comment?

Reply:
I personally agree with the current behaviour. There's no point feeding the SPIR-V to the implementation under test if the external toolchain produced an invalid module (except for negative tests but these wouldn't use the offline compilation framework anyway).

Reply:
This behavior is fine so long as the SPIR-V LLVM translator generates valid SPIR-V and the SPIR-V validator for OpenCL is up-to-date and doesn't have false-positives, but frankly I don't have that level of confidence in either component yet.

I'm not going to lose a ton of sleep over this because there is an option to disable the validator, but it'd be a shame to add these changes to run the validator automatically only to require disabling the validator to run any meaningful tests to completion.

Reply:
I agree with your assessment of the current quality level of the tools. Part of the reason for enabling this by default is to drive improvements to the SPIR-V LLVM Translator and SPIR-V validator.